mmgen/datasets/pipelines/augmentation.py

```diff
@@ -205,7 +205,7 @@ def __call__(self, results):
                 scale = (self.scale[-1], int(self.scale[-1] / w * h))
         else:
             # direct use the given ones
-            scale = self.scale
+            scale = self.scale[::-1]
```

Comment:
We should consider the situation when the input scale is float or int.

Reply:
I don't see the `mmcv.imresize` can take argument `scale` of `float` or `int`.
https://github.com/open-mmlab/mmcv/blob/51b40c332aff9d2927fcc252b248d295850a4d55/mmcv/image/geometric.py#L51-L95

Reply:
I think the original `Resize` pipeline in the `mmgen` does NOT support taking as input `scale` of `float` or `int`. I tried and got an error raised.

Reply:
However, `mmcv.imrescale` can take `float` as input. In MMGen's `Resize`, we compose `mmcv.imrescale` and `mmcv.imresize` in one function. Refers to
https://github.com/open-mmlab/mmgeneration/blob/69333cfa4b41768b9d700d2a913e3cfa567e1c9e/mmgen/datasets/pipelines/augmentation.py#L162-L177

Reply:
You may use the following command to run the unit test locally.
```
 coverage run --branch --source mmgen -m pytest -s tests/test_datasets/test_pipelines/test_augmentation.py
```

Reply:
I see... I think it's a little bit confusing though. Current `Resize`:
| Cases |Arguments| Comments|
| -----------| ------| ---- |
|scale with a factor & keep ratio | scale=float, keep_ratio=True| lack of asserting keep_ratio=True; not support scale=int|
|scale with factors (fh, fw) & not keep ratio | not supported| |
|scale with size (h, w) & not keep ratio | scale=(h, w), keep_ratio=False| Misplaced `h` and `w`|
scale with size (h, -1) & keep ratio | scale=(h, -1), keep_ratio=True;| lack of asserting keep_ratio=True; variable `max_long_edge` could be the actual short edge|

Reply:
I can help resolve the problems:
1. lacking asserting `keep_ratio`=True;
2. not support scale=int;
3. Misplaced `h` and `w` (solved).

But as for the " variable `max_long_edge` could be the actual short edge", it requires futher discussion.